backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/PredicatesHandler.java

```diff
@@ -176,7 +176,7 @@
 
     switch (value.getKindCase()) {
       case NUMBER_VALUE:
-        sql += applyOperator(operator, columnAsNumber(colValue, true), ":" + valueBindingName);
+        sql += applyOperator(operator, colValue, ":" + valueBindingName);
```

Comment:
Removed casting in number because while doing casting it will not return result so after testing and spending some time i realize that the query is working without casting as expected.